tests/Benchmarks/HttpParserBench.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+using System.Text.Http;
+using System.Text.Http.Parser;
+
+public class HttpParserBench
+{
+    private static readonly byte[] s_plaintextTechEmpowerRequestBytes = Encoding.UTF8.GetBytes(_plaintextTechEmpowerRequest);
+
+    private const string _plaintextTechEmpowerRequest =
+        "GET /plaintext HTTP/1.1\r\n" +
+        "Host: localhost\r\n" +
+        "Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7\r\n" +
+        "Connection: keep-alive\r\n" +
+        "\r\n";
+
+    [Benchmark(InnerIterationCount = 1000)]
+    static void HttpParserRob()
+    {
+        ReadableBuffer buffer = ReadableBuffer.Create(s_plaintextTechEmpowerRequestBytes);
+        var parser = new HttpParser();
+        var request = new Request();
+
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            using (iteration.StartMeasurement())
+            {
+                parser.ParseRequestLine(request, buffer, out var consumed, out var read);
+            }
+        }
+    }
+
+    [Benchmark(InnerIterationCount = 1000)]
+    static void HttpParserReadableBytes()
+    {
+        var buffer = new ReadOnlyBytes(s_plaintextTechEmpowerRequestBytes);
+        var parser = new HttpParser();
+        var request = new Request();
+
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            using (iteration.StartMeasurement())
+            {
+                parser.ParseRequestLine(request, buffer, out var consumed);
+            }
+        }
+    }
+
+    [Benchmark(InnerIterationCount = 1000)]
```

Comment:
I wasn't aware of this. I thought you didn't have control over Benchmark.Iterations and that attribute set the InnerIterationCount for loop bodies that take too little time to execute.

https://github.com/Microsoft/xunit-performance#authoring-benchmarks

> In addition, you can add inner iterations to the code to be measured.
Add the for loop using Benchmark.InnerIterationCount as the number of loop iterations
Specify the value of InnerIterationCount using the [Benchmark] attribute